src/libraries/System.Private.CoreLib/src/System/Environment.Win32.cs

```diff
@@ -217,31 +217,40 @@ private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOptio
             // The only SpecialFolderOption defines we have are equivalent to KnownFolderFlags.
 
             string folderGuid;
-
+            string? fallbackEnv = null;
             switch (folder)
             {
                 // Special-cased values to not use SHGetFolderPath when we have a more direct option available.
                 case SpecialFolder.System:
-                    // This assumes the system directory always exists and thus we don't need to do anything special for any SpecialFolderOption.
-                    return SystemDirectory;
+                    if (!string.IsNullOrEmpty(SystemDirectory))
+                        return SystemDirectory;
+
+                    string? windir = Environment.GetEnvironmentVariable("windir");
```

Comment:
Is this fallback needed? I would expect that `SystemDirectory` is going to work fine on Nano Server today. If it is really needed, should it better be in the SystemDirectory?